src/main/java/io/mosip/mimoto/controller/V2IssuersController.java

```diff
@@ -0,0 +1,163 @@
+
+    @Autowired
+    IdpService idpService;
+
+    @Autowired
+    IssuersService issuersService;
+
+    @Value("${mosip.oidc.esignet.token.endpoint}")
+    String tokenEndpoint;
+
+    @GetMapping
+    public ResponseEntity<Object> getAllIssuers(@RequestParam(required = false) String search){
+        ResponseWrapper<IssuersDTO> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v2");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+
+        try {
+            responseWrapper.setResponse(v2IssuersService.getAllIssuers(search));
+        } catch (ApiNotAccessibleException | IOException e) {
+            logger.error("Exception occurred while fetching issuers ", e);
+            responseWrapper.setErrors(List.of(new ErrorDTO(API_NOT_ACCESSIBLE_EXCEPTION.getCode(), API_NOT_ACCESSIBLE_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+        }
+
+        return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
+    }
+
+    @GetMapping("/{issuer-id}/credentials-supported")
+    public ResponseEntity<Object> getCredentialsSupportedForIssuer(@PathVariable("issuer-id") String issuerId,
+                                                                   @RequestParam(required = false) String search) {
+        ResponseWrapper<Object> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v2");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+        IssuerSupportedCredentialsResponse credentialsSupported;
+        try {
+            credentialsSupported = v2IssuersService.getCredentialsSupported(issuerId, search);
+        }catch (ApiNotAccessibleException | IOException exception){
+            logger.error("Exception occurred while fetching issuers ", exception);
+            responseWrapper.setErrors(List.of(new ErrorDTO(API_NOT_ACCESSIBLE_EXCEPTION.getCode(), API_NOT_ACCESSIBLE_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseWrapper);
+        }
+        responseWrapper.setResponse(credentialsSupported);
+
+        if (credentialsSupported.getSupportedCredentials().isEmpty()) {
+            logger.error("invalid issuer id passed - {}", issuerId);
+            responseWrapper.setErrors(List.of(new ErrorDTO(INVALID_ISSUER_ID_EXCEPTION.getCode(), INVALID_ISSUER_ID_EXCEPTION.getMessage())));
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(responseWrapper);
+        }
+
+        return ResponseEntity.status(HttpStatus.OK).body(responseWrapper);
+    }
+
+    @PostMapping(value = "/get-token/issuer", consumes = {MediaType.APPLICATION_FORM_URLENCODED_VALUE})
+    public ResponseEntity getToken(@RequestParam Map<String, String> params, @RequestParam String issuer) {
```

Comment:
If we are sending issuer as request param then, there is no need to have `/issuer` as part of Api url

Reply:
resolved